src/software/ai/hl/stp/play/offense/offense_play.cpp

```diff
@@ -26,20 +26,30 @@ void OffensePlay::getNextTactics(TacticCoroutine::push_type &yield, const World
 
 void OffensePlay::updateTactics(const PlayUpdate &play_update)
 {
-    PriorityTacticVector tactics_to_return;
-    unsigned int num_shoot_or_pass = play_update.num_tactics - 2;
-    unsigned int num_defenders     = 2;
     if (play_update.num_tactics == 0)
     {
         return;
     }
-    if (play_update.num_tactics <= 3)
+
+    PriorityTacticVector tactics_to_return;
+    unsigned int num_defenders = 2;
+    unsigned int num_enemy_robots =
+        static_cast<int>(play_update.world.enemyTeam().numRobots());
+
+    if (num_enemy_robots <= 3)
+    {
+        num_defenders = 1;
+        if (num_enemy_robots < 2)
+            num_defenders--;
```

Comment:
doesn't this mean we just have 0 defender robots when there are less than two enemy robots, may as well just set it to 0 to be clearer